connection/cluster/ClusterDatabaseManager.java

```diff
@@ -59,25 +59,58 @@ public ClusterDatabaseManager(ClusterClient client) {
 
     @Override
     public boolean contains(String name) {
-        return failsafeTask(name, (stub, dbMgr) -> dbMgr.contains(name));
+        try {
+            BiFunction<ClusterServerStub, TypeDBDatabaseManagerImpl, Boolean> task = ((stub, dbMgr) -> dbMgr.contains(name));
+            return client.createFailsafeTask(
+                    name,
+                    parameter -> task.apply(
+                            parameter.client().stub(),
+                            parameter.client().databases()
+                    )
+            ).runPrimaryReplica();
+        } catch (TypeDBClientException e) {
+            if (e.getErrorMessage().equals(DB_DOES_NOT_EXIST)) return false;
+            else throw e;
+        }
```

Comment:
Gets rid of the failSafeTask wrapper function (below in red). The wrapper would try running the task against all replicas. If it then receives a NOT_PRIMARY_REPLICA` exception, it retries on the primary replica.

We update each task to go to be more explicit in where it should run.

Reply:
Unsure about doing any running within the function. For brevity, I've reintroduced `failSafeTask` which only creates it but does no running. Please re-review.

Reply:
That's even better, nicely done.